server/controllers/eventsController.ts

```diff
@@ -0,0 +1,137 @@
+import { Request, Response } from 'express';
+import { Event } from 'server/models/Event';
+
+export default {
+  async index(req: Request, res: Response) {
+    const { chapter_id } = req.params;
+
+    const events = await Event.find({ where: { chapter: chapter_id } });
+
+    res.json(events);
+  },
+
+  async show(req: Request, res: Response) {
+    const { id, chapter_id } = req.params;
+
+    const event = await Event.findOne({
+      where: {
+        id: parseInt(id),
+        chapter: chapter_id,
+      },
+    });
+
+    if (event) {
+      res.json(event);
+    } else {
+      res.status(404).json({ error: 'Cant find event' });
+    }
+  },
+
+  async create(req: Request, res: Response) {
+    const {
+      name,
+      chapter,
+      description,
+      capacity,
+      venue,
+      canceled,
+      start_at,
+      ends_at,
+    } = req.body;
+
+    const event = new Event({
+      name,
+      chapter,
+      description,
+      capacity,
+      venue,
+      canceled,
+      start_at,
+      ends_at,
+    });
+
+    try {
+      await event.save();
+      res.status(201).json(event);
+    } catch (e) {
+      if (e.code === '23503' && e.message.includes('foreign key constraint')) {
+        if (e.detail.includes('venue')) {
+          return res.status(404).json({ message: 'venue not found' });
+        }
+        if (e.detail.includes('chapter')) {
+          return res.status(404).json({ message: 'chapter not found' });
+        }
+      }
+
+      res.status(500).json({ error: e });
+    }
+  },
+
+  async update(req: Request, res: Response) {
+    const { id, chapter_id } = req.params;
+    const {
+      name,
+      //   chapter,
+      description,
+      capacity,
+      venue,
+      canceled,
+      start_at,
+      ends_at,
+    } = req.body;
+
+    const event = await Event.findOne({
+      where: { id: parseInt(id), chapter: chapter_id },
+    });
+
+    if (event) {
+      name && (event.name = name);
+      //   chapter && (event.chapter = chapter);
+      description && (event.description = description);
+      capacity && (event.capacity = capacity);
```

Comment:
This breaks if/when capacity is 0.

Reply:
Another thought, this is a prime candidate for [nullish coalescing](https://github.com/tc39/proposal-nullish-coalescing). 🤐